### ISSUE TITLE
Uncaught Error: Cannot decorate a destroyed marker

### ISSUE THREAD

**Issue by danielvos** [victorhqc/color-indent#3]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.21.1 x64
**Electron**: 1.6.15
**OS**: Microsoft Windows 10 Pro
**Thrown From**: [color-indent](https://github.com/victorhqc/color-indent) package 0.5.2

### Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At C:\Users\Daniel\Desktop\ATOM 1\atom-x64-windows\Atom x64\resources\app\src\decoration-manager.js:191

Error: Cannot decorate a destroyed marker
    at DecorationManager.decorateMarker (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/decoration-manager.js:183:27)
    at TextEditor.module.exports.TextEditor.decorateMarker (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/text-editor.js:1811:43)
    at paintTextEditorLine (/packages/color-indent/lib/color-indent-render.js:81:14)
    at paintMultipleLines (/packages/color-indent/lib/color-indent-render.js:104:5)
    at /packages/color-indent/lib/color-indent-render.js:128:7
    at arrayMap (/packages/color-indent/node_modules/lodash/_arrayMap.js:16:21)
    at map (/packages/color-indent/node_modules/lodash/map.js:50:10)
    at textDidChange (/packages/color-indent/lib/color-indent-render.js:114:37)
    at wrapper (/packages/color-indent/node_modules/lodash/_createHybrid.js:87:15)
    at Function.module.exports.Emitter.simpleDispatch (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at TextBuffer.module.exports.TextBuffer.emitDidChangeTextEvent (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/node_modules/text-buffer/lib/text-buffer.js:1675:28)
    at TextBuffer.module.exports.TextBuffer.transact (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/node_modules/text-buffer/lib/text-buffer.js:975:18)
    at TextEditor.module.exports.TextEditor.transact (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/text-editor.js:1673:32)
    at HTMLElement.newCommandListeners.(anonymous function) (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/register-default-commands.js:676:28)
    at CommandRegistry.handleCommandEvent (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/command-registry.js:381:36)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:621:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:412:28)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (~/Desktop/ATOM 1/atom-x64-windows/Atom x64/resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
     -0:45.7.0 application:open-your-stylesheet (atom-text-editor.editor.is-focused)
     -0:39.3.0 core:move-up (input.hidden-input)
  2x -0:38.9.0 editor:newline (input.hidden-input)
     -0:37.9.0 core:move-up (input.hidden-input)
  2x -0:30.4.0 core:backspace (input.hidden-input)
  2x -0:29.2.0 editor:newline (input.hidden-input)
  2x -0:28.6.0 core:backspace (input.hidden-input)
     -0:23.3.0 core:confirm (button.btn.btn-default.tool-bar-btn.ion.ion-android-document)
  2x -0:21 editor:newline (input.hidden-input)
  2x -0:20.3.0 core:backspace (input.hidden-input)
  2x -0:19.4.0 editor:newline (input.hidden-input)
     -0:14.2.0 core:backspace (input.hidden-input)
     -0:10.9.0 editor:newline (input.hidden-input)
  3x -0:10 core:backspace (input.hidden-input)
     -0:08.7.0 editor:newline (input.hidden-input)
  3x -0:08.3.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
ascii-table 0.1.1 
atom-clock 0.1.15 
atom-jump-block 0.5.4 
atom-show-project-name 0.1.4 
auto-soft-wrap 0.3.1 
browser-plus 0.0.98 
busy-signal 1.4.3 
click-link 1.0.11 
clipboard-plus 0.5.1 
color-indent 0.5.2 
color-picker 2.2.5 
directory-color 0.3.0 
file-icons 2.1.13 
filesize 3.0.0 
flex-tool-bar 0.12.0 
full-width-tabs 0.3.0 
gpp-compiler 3.0.7 
hey-pane 1.0.0 
highlight-column 0.5.4 
highlight-selected 0.13.1 
intentions 1.1.5 
language-ini 1.19.0 
language-latex 1.1.1 
latex 0.48.0 
latex-friend 0.0.11 
line-count-status 0.0.3 
linter 2.2.0 
linter-chktex 1.3.1 
linter-gcc 0.7.1 
linter-ui-default 1.6.10 
minimap 4.29.7 
minimap-bookmarks 0.4.2 
minimap-codeglance 0.4.7 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-highlight-selected 4.6.1 
minimap-linter 2.1.1 
minimap-pigments 0.2.2 
minimap-selection 4.5.0 
open-recent 5.0.0 
outline-selection 0.3.2 
pdf-view 0.59.0 
pigments 0.40.2 
pinned-tabs 1.3.0 
platformio-ide-terminal 2.7.0 
preview-inline 1.4.7 
scroll-color 0.3.0 
simple-drag-drop-text 0.3.6 
symbols-tree-view 0.14.0 
sync-settings 0.8.3 
tab-switcher 1.5.6 
title-bar-replacer 1.7.0 
tool-bar 1.1.0 
tree-view-autoresize 2.2.0 
wordcount 2.10.4 
```

**Comment by victorhqc** [comment]
Thanks for reporting this. I'll look into it 👍

**Comment by victorhqc** [comment]
@danielvos can you try to replicate the issue with `atom --safe` I'm unable to replicate this issue so far

**Comment by AgtLucas** [comment]
I had the same problem a few hours ago, I just reopened the current file and it was fine. 😄

**Comment by victorhqc** [comment]
Published a new version `0.5.3` although not sure if that'll fix the issue
